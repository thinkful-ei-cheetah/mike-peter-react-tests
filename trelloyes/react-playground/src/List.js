import React from 'react';
import Card from './Card';


function List (props) {
    const allCards = props.store.allCards;

    const listCards = props.cardIds.map( id => 
        <Card title={allCards[id].title} content={allCards[id].content} />
        );


    return (
        <section id={props.id} className="List">
        <header className="List-header">
            <h2>{props.header}</h2>
       </header>

       <div className="List-cards">
           <div>
           {listCards}
           {/* <Card title={allCards.a.title} content={allCards.a.content} /> */}
           </div>

       </div>
           
     </section>
        );
}


export default List;