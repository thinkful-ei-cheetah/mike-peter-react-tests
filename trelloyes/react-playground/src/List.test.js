import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import STORE from './store.js';

//this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  //render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<List id={STORE.lists[0].id} header={STORE.lists[0].header} cardIds={STORE.lists[0].cardIds} store={STORE} />, div);

  //clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<List id={STORE.lists[0].id} header={STORE.lists[0].header} cardIds={STORE.lists[0].cardIds} store={STORE} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});

it('renders the UI as expected with no unreads', () => {
  const tree = renderer
    .create(<List id={STORE.lists[0].id} header={STORE.lists[0].header} cardIds={STORE.lists[0].cardIds} store={STORE} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});