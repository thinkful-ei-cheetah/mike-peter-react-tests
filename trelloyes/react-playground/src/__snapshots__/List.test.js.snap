// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
  id="1"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div>
      <div
        className="card"
      >
        <h3>
          First card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Second card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Fifth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Sixth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Seventh card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Tenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Twelth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Thirteenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders the UI as expected with no unreads 1`] = `
<section
  className="List"
  id="1"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div>
      <div
        className="card"
      >
        <h3>
          First card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Second card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Fifth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Sixth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Seventh card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Tenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Twelth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
      <div
        className="card"
      >
        <h3>
          Thirteenth card
        </h3>
        <p>
          lorem ipsum
        </p>
      </div>
    </div>
  </div>
</section>
`;
